test/cpp/api/functional.cpp

```diff
@@ -575,6 +575,80 @@ TEST_F(FunctionalTest, LogSigmoid) {
   ASSERT_TRUE(torch::allclose(y, y_exp, 1e-4, 1e-7));
 }
 
+TEST_F(FunctionalTest, GumbelSoftmax) {
+
+    // Test 1: 1D shape, 0 and -1 dim
```

Comment:
nit: we should indent by 2 spaces for all changes in this PR, because the rest of the codebase does so :D

Reply:
I've done that everywhere now